animation_nodes/nodes/gpencil/gp_object_input.py

```diff
@@ -0,0 +1,111 @@
+import bpy
+from bpy.props import *
+from ... events import propertyChanged
+from ... base_types import AnimationNode, VectorizedSocket
+from ... data_structures import GPLayer, GPFrame, GPStroke, DoubleList, Vector3DList
+
+layerTypeItems = [
+    ("ALL", "All", "Get all grease pencil layers", "NONE", 0),
+    ("SINGLE", "Single", "Get a specific grease pencil layer", "NONE", 1)
+]
+
+class GPObjectInputNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_GPObjectInputNode"
+    bl_label = "GP Object Input"
+    errorHandlingType = "EXCEPTION"
+
+    layerType: EnumProperty(name = "Layer Type", default = "SINGLE",
+        items = layerTypeItems, update = AnimationNode.refresh)
+
+    def create(self):
+        self.newInput("Object", "Object", "object", defaultDrawType = "PROPERTY_ONLY")
+        self.newInput("Boolean", "Use World Space", "useWorldSpace", value = True)
+
+        if self.layerType == "SINGLE":
+            self.newInput("Text", "Layer Name", "layerName")
+            self.newOutput("GPLayer", "Layer", "layer")
+        elif self.layerType == "ALL":
+            self.newOutput("GPLayer List", "Layers", "layers")
+
+    def draw(self, layout):
+        layout.prop(self, "layerType", text = "")
+
+    def getExecutionFunctionName(self):
+        if self.layerType == "SINGLE":
+            return "executeSingle"
+        elif self.layerType == "ALL":
+            return "executeList"
+
+    def executeSingle(self, object, useWorldSpace, layerName):
+        if object is None:
+            return GPLayer()
+
+        if useWorldSpace:
+            worldMatrix = object.matrix_world
```

Comment:
You have to get the matrix from an evaluated object. See 081819d for how to do that.

Reply:
It is done.